plugin/src/index.ts

```diff
@@ -58,35 +58,49 @@ export async function onBuild({
 The plugin no longer uses this and it should be deleted to avoid conflicts.\n`)
   }
 
-  if (shouldSkipFunctions(cacheDir)) {
-    await deleteFunctions(constants)
-    return
-  }
-  const compiledFunctionsDir = path.join(cacheDir, '/functions')
+  const neededFunctions = await getNeededFunctions(cacheDir)
 
-  await writeFunctions({ constants, netlifyConfig })
+  await deleteFunctions(constants)
 
-  mutateConfig({ netlifyConfig, cacheDir, compiledFunctionsDir })
+  const compiledFunctionsDir = path.join(cacheDir, '/functions')
 
-  const root = dirname(netlifyConfig.build.publish)
-  await patchFile(root)
-  await relocateBinaries(root)
+  await writeFunctions({ constants, netlifyConfig, neededFunctions })
 
-  // Editing _redirects so it works with ntl dev
-  spliceConfig({
-    startMarker: '# @netlify/plugin-gatsby redirects start',
-    endMarker: '# @netlify/plugin-gatsby redirects end',
-    contents: '/api/* /.netlify/functions/__api 200',
-    fileName: join(netlifyConfig.build.publish, '_redirects'),
+  mutateConfig({
+    netlifyConfig,
+    cacheDir,
+    compiledFunctionsDir,
+    neededFunctions,
   })
+
+  if (neededFunctions.includes('DSG')) {
```

Comment:
@ascorbic Do you know if this conditional should test for DSG _and_ SSR? I'm assuming that DSG requires the `patchFile` and `relocateBinaries`, but I'm not sure if SSR does without more testing.

Reply:
Yes, they both need it